src/Sylius/Component/Product/spec/Model/ProductSpec.php

```diff
@@ -204,6 +204,50 @@ function its_hasVariants_should_return_true_only_if_any_variants_defined(Variant
     function it_should_initialize_variants_collection_by_default()
     {
         $this->getVariants()->shouldHaveType('Doctrine\Common\Collections\Collection');
+        $this->getAvailableVariants()->shouldHaveType('Doctrine\Common\Collections\Collection');
+    }
+
+    function it_should_exclude_unavailable_variants_when_available_variants_are_called(VariantInterface $variant)
+    {
+        $variant->isMaster()->willReturn(false);
+        $variant->isAvailable()->willReturn(false);
+
+        $variant->setProduct($this)->shouldBeCalled();
+
+        $this->addVariant($variant);
+        $this->getAvailableVariants()->shouldHaveCount(0);
+    }
+
+    function it_should_exclude_master_variants_when_available_variants_are_called(VariantInterface $variant)
+    {
+        $variant->isMaster()->willReturn(true);
+
+        $variant->setProduct($this)->shouldBeCalled();
+
+        $this->addVariant($variant);
+        $this->getAvailableVariants()->shouldHaveCount(0);
+    }
+
+    function it_should_return_available_variants_when_available_variants_are_called(
```

Comment:
I just followed the method names of the existing tests. Contains a bunch of methods including "should". But thanks for the explanation.